src/d3-ext/profile.js

```diff
@@ -0,0 +1,320 @@
+      gEnter.style('font', '11px Arial');
+      gEnter.append('path').attr('class', 'area')
+          .style('fill', FILL_COLOR);
+      gEnter.append('path').attr('class', 'line')
+          .style('stroke', STROKE_COLOR)
+          .style('fill', 'none');
+
+      gEnter.insert('g', ':first-child')
+          .attr('class', 'grid-y');
+
+      gEnter.append('g')
+          .attr('class', 'x axis')
+          .attr('transform', 'translate(0,' + height + ')');
+
+      gEnter.append('text')
+          .attr('class', 'x label')
+          .attr('text-anchor', 'end')
+          .attr('x', width - 4)
+          .attr('y', height - 4);
+
+      gEnter.append('g')
+          .attr('class', 'y axis');
+
+      gEnter.append('text')
+          .attr('class', 'y label')
+          .attr('text-anchor', 'end')
+          .attr('y', 6)
+          .attr('dy', '.75em')
+          .attr('transform', 'rotate(-90)')
+          .style('fill', 'grey')
+          .text('elevation (m)');
+
+      gEnter.append('g')
+          .attr('class', 'metas')
+          .attr('transform', 'translate(' + (width + 3) + ', 0)');
+
+      var yHover = gEnter.append('g').attr('class', 'y grid-hover');
+      yHover.append('svg:line').attr('stroke-dasharray', '5,5');
+      yHover.append('text');
+
+      var xHover = gEnter.append('g').attr('class', 'x grid-hover');
+      xHover.append('svg:line').attr('stroke-dasharray', '5,5');
+      xHover.append('text');
+
+      gEnter.append('rect')
+          .attr('class', 'overlay')
+          .attr('width', width)
+          .attr('height', height)
+          .style('fill', 'none')
+          .style('pointer-events', 'all');
+
+      // Update the outer dimensions.
+      svg
+          .attr('width', width + margin.left + margin.right)
+          .attr('height', height + margin.top + margin.bottom);
+
+      // Update the inner dimensions.
+      var g = svg.select('g')
+          .attr('transform', 'translate(' + margin.left + ',' +
+              margin.top + ')');
+
+      var xDomain = d3.extent(data, function(d) { return d['dist']; });
+      x.domain(xDomain);
+
+      var yDomain = [d3.min(data, function(d) { return d['values']['mnt']; }),
+            d3.max(data, function(d) { return d['values']['mnt']; })];
+
+      var padding = (yDomain[1] - yDomain[0]) * 0.1;
+      y.domain([yDomain[0] - padding, yDomain[1] + padding]);
+
+      // Update the area path.
+      g.select('.area')
+          .transition()
+          .attr('d', area);
+      g.select('.line')
+          .transition()
+          .attr('d', line);
+
+      if (xDomain[1] > 2000) {
+        xFactor = 1000;
+        units = 'km';
+      } else {
+        xFactor = 1;
+        units = 'm';
+      }
+
+      xAxis.tickFormat(function(d) {
+        return d / xFactor;
+      });
+
+      g.select('.x.axis')
+          .transition()
+          .call(xAxis);
+
+      g.select('.x.label')
+          .text('distance (' + units + ')')
```

Comment:
Need translation, see https://github.com/camptocamp/ngeo/pull/173/files#r25350768
In this case we could pass a parameter to the translator.